python/pyspark/pipelines/cli.py

```diff
@@ -0,0 +1,276 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+"""
+Implementation of spark-pipelines CLI.
+
+Example usage:
+    $ bin/spark-pipelines run --spec /path/to/pipeline.yaml
+"""
+from contextlib import contextmanager
+import argparse
+import importlib.util
+import os
+import yaml
```

Comment:
Seems like the PySpark tests fail if `yaml` is not installed (https://github.com/apache/spark/actions/runs/15516895252/job/43685059103). I think we should skip the tests if yaml is not found.

Reply:
Is this what we do for other dependencies? I'd be worried that, if we accidentally make a change to Spark CI that avoids installing pyyaml when we'd otherwise expect it to be installed, then the tests could get broken and we wouldn't find out.

I could alternatively help trace down why it isn't installed in that situation and fix it?

Reply:
Yeah we do .. the tests work without any dependency basically. That build is a scheduled build dedicated for that (to test without any dependencies).